### ISSUE TITLE
About loss

### ISSUE THREAD

**Issue by yyl199655** [apache/incubator-mxnet#11385]
If I want to add other loss function or penalty term, how to implement it?,Do I need to writer the BackPropagation

**Comment by vrakesh** [comment]
Hi @yyl199655 thanks for the question, @sandeep-krishnamurthy requesting this be labeled under Question

**Comment by szha** [comment]
@yyl199655 if you use gluon, you can just write whatever penalty term you want using ndarray interface inside training loop. Only forward computation is needed, as long as the function is differentiable

**Comment by ThomasDelteil** [comment]
@yyl199655 for more information automatic differentiation (autograd) you can follow the Gluon Crash Course, in particular this [chapter](https://gluon-crash-course.mxnet.io/autograd.html)

To expand on @szha explanation, you simply need to compute your loss function within the `with autograd.record():` scope and then you can call `loss.backward()` on your loss term to automatically compute that loss with respect to the parameters that were used to compute it.

@yyl199655 if you would want to follow up, please open a thread in https://discuss.mxnet.io, Thanks
@szha could you please close the issue? Thanks!

**Comment by yyl199655** [comment]
If I want to extract a few lines of a matrix(symbol), how should I do?
